src/main/java/it/pagopa/ecommerce/payment/methods/controller/v1/PaymentMethodsController.java

```diff
@@ -99,11 +99,7 @@ public ResponseEntity<ProblemJsonDto> errorHandler(RuntimeException exception) {
                     HttpStatus.CONFLICT
             );
         } else if (exception instanceof SessionAlreadyAssociatedToTransaction) {
-            return new ResponseEntity<>(
-                    new ProblemJsonDto().status(409).title("Session already associated to transaction")
-                            .detail(exception.getMessage()),
-                    HttpStatus.CONFLICT
-            );
+            return ResponseEntity.noContent().build();
```

Comment:
If this was the only case that generated a 409 please update the api spec :)

Reply:
Also I think the spec needs updating because 204 is not included in the response codes in the API

Reply:
What about update Redis document TTL here? If the retry is triggered by user (by click on the payment button again after some minute) TTL document will remain the same of the first attempt.

This can be achieved performing a new save operation on Redis with the same document and key or by the SET EX Redis command (that I think should be add to the [ecommerce-commons RedisTemplateWrapper implementation](https://github.com/pagopa/pagopa-ecommerce-commons/blob/main/src/main/java/it/pagopa/ecommerce/commons/redis/templatewrappers/RedisTemplateWrapper.java))

Reply:
@pietro-tota Do you mean performing the save on Redis also on retry operation without throwing (and then handling) the SessionAlreadyAssociatedToTransaction exception?

Reply:
Yes, this can be one way of achieve this. We can perform save operation on the same document multiple times and this will have as result that also TTL is updated too here. But removing checks against transactionId/orderId association may lead to transactionId to be overridden here.
Taking that into account I would leave check as is and put here only a SET-EX command here to update already existing document TTL only

Reply:
One more think, I would avoid response modification here in the exception handler method.
It would make code clearer to properly handle this error at controller/service method level and avoid throw `SessionAlreadyAssociatedToTransaction` exception at all for those cases and leave exception handler method as is (in case of this exception re-use in future)

Reply:
@giovanniberti api spec updated in 863ad4c

Reply:
@pietro-tota I don't know if keeping on extending the document TTL on every retry is a good idea, what if it keeps getting extended indefinitely? What's the goal of updating the document TTL?

Reply:
> One more think, I would avoid response modification here in the exception handler method. It would make code clearer to properly handle this error at controller/service method level and avoid throw `SessionAlreadyAssociatedToTransaction` exception at all for those cases and leave exception handler method as is (in case of this exception re-use in future)

It is a fair point but as far as I understood that exception is used only by that api and it has been intended to model the case of a transaction that has been already bound to a session by updating the transaction id on the session. Since our goal is to change how the api behaves in this case (someone is trying again to associate the transaction id to that session), I would keep the same exception and change how the controller is handling it to achieve our goal. Since this exception models this very specific case, I don't some other api will use it in the future.

Reply:
Proposed changes applied with fa80f8d